syntax = "proto3";

service InferenceService {
  rpc InitializeSession (SessionInitRequest) returns (SessionInitResponse);
  rpc SendInference (InferenceRequest) returns (InferenceResponse);
}

message SessionInitRequest {
  int32 num_requests = 1;
}

message SessionInitResponse {
  string message = 1;
}

message InferenceRequest {
  int32 id = 1;
  bytes tensor = 2;
}

message InferenceResponse {
  int32 id = 1;
  string result = 2;
}
